### ISSUE TITLE
[Research] OUI Compliance audit of the `vis_type_tagcloud` plugin

### ISSUE THREAD

**Issue by BSFishy** [opensearch-project/OpenSearch-Dashboards#4100]
Tracking issue for OUI compliance audit of the `vis_type_tagcloud` plugin.

Follow the steps in #3945 to complete the audit, then open another issue with findings. Once the work of performing the audit is done, this issue can be closed.

<details>
  <summary>Related files</summary>

  size (bytes) | file | notes
---|---|---
442 | [src/plugins/vis_type_tagcloud/public/components/tag_cloud.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/vis_type_tagcloud/public/components/tag_cloud.scss) | 

</details>

If this issue isn't already assigned to someone and you'd like to take on this work, please ping @BSFishy in the comments and I can assign it to you.

**Comment by nurSaadat** [comment]
[tag_cloud.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/vis_type_tagcloud/public/components/tag_cloud.scss) has four custom classes.

### tgcChart__container and tgcChart__wrapper 
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/20c928fccf9b2516b36613874b335e9291b69f81/src/plugins/vis_type_tagcloud/public/components/tag_cloud.scss#L8-L12

They are used in `tag_cloud_chart.tsx` to wrap a tag cloud component.
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/20c928fccf9b2516b36613874b335e9291b69f81/src/plugins/vis_type_tagcloud/public/components/tag_cloud_chart.tsx#L83-L89

I suggest replacing `div className="tgcChart__wrapper"` by `EuiFlexGroup`.
Also, I suggest replacing `div className="tgcChart__container"` by `EuiFlexItem`. However, I found that `EuiFlexItem` doesn't have property `ref`, so maybe it can be added.

### tgcChart 
This class is used in [tag_cloud_visualization.js](src/plugins/vis_type_tagcloud/public/components/tag_cloud_visualization.js)
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/20c928fccf9b2516b36613874b335e9291b69f81/src/plugins/vis_type_tagcloud/public/components/tag_cloud_visualization.js#L50-L54

I suggest element creation should be realized as a function.

### tgcChart__labelhttps://github.com/opensearch-project/OpenSearch-Dashboards/blob/f017eaacd97a7a575bc9b0d34d9e919caf152b32/src/plugins/vis_type_tagcloud/public/components/tag_cloud.scss#L23-L27

This class is used to style the label of a tag cloud. Apparently, this label is not shown, as the property `shouldShowLabel` is never changed.

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/f017eaacd97a7a575bc9b0d34d9e919caf152b32/src/plugins/vis_type_tagcloud/public/components/label.js#L33-L49

Later, `Label` component is used as follows
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/f017eaacd97a7a575bc9b0d34d9e919caf152b32/src/plugins/vis_type_tagcloud/public/components/tag_cloud_visualization.js#L92-L95

I suggest `Label` component can be substituted by OUI text component.

### Conclusion
* Add `ref` property to `EuiFlexItem`
* Substitute custom classes by OUI components
* Refactor new DOM element creation

**Comment by nurSaadat** [comment]
@joshuarrrr @BSFishy could you assign me to this issue?

**Comment by joshuarrrr** [comment]
@nurSaadat Do you mind moving your audit findings to a new issue? Sorry for the extra administrative overhead, but it will help us track progress on the overall initiative easier.

**Comment by nurSaadat** [comment]
@joshuarrrr sure
